src/strands/models/bedrock.py

```diff
@@ -443,7 +445,24 @@ def _stream(
                             for event in self._generate_redaction_events():
                                 callback(event)
 
-                    callback(chunk)
+                    # Track if we see tool use events
+                    if "contentBlockStart" in chunk and chunk["contentBlockStart"].get("start", {}).get("toolUse"):
+                        has_tool_use = True
+
+                    # Fix stopReason for streaming responses that contain tool use
+                    if "messageStop" in chunk and has_tool_use:
+                        message_stop = chunk["messageStop"]
+                        if message_stop.get("stopReason") == "end_turn":
```

Comment:
small nit; these if statement can be combined I think to reduce the branching logic.

E.g. this could be:

```python
if has_tool_use and "messageStop" in chunk and (message_stop := chunk["messageStop"]).get("stopReason") == "end_turn":
```